### DISCUSSION TITLE
Missing funds

### DISCUSSION THREAD

**Discussion by barfinger** [BlitzWallet/BlitzWallet#45] [Q&A]
Blitz 0.3.0 from zap.store

I sent a "successful payment" from Blitz to Phoenix with no payment preimage. My balance in the app was almost completely missing too(2x more missing than I sent) About 13 days later I got it all back in a channel close on chain.

I have since tried to send from CLN via Zeus _to Blitz_. Zeus produced an error(as it generally does over Tor) and my payments in CLN show a _successful payment_ with a payment preimage. But _no funds arrived in Blitz_.

- What is going on? 
- Is this common with lightning, I haven't had this issue with anything else?
- Why did Blitz send appear to send or lose so much of my funds successfully without a preimage and then take ~13 days to send everything back on chain?
- Will my funds magically come back again to my CLN since Blitz never received it?
  - How or Who do I contact to resolve stuck funds going to my Blitz wallet?

Excuse my ignorance, but nothing about lightning works how you expect, and getting the answers isn't exactly easy.

I have recently installed Blitz 0.3.4.

**Comment by BlakeKaufman** [comment]
A couple of questions. 

1) When you have been sending transactions did you open a lightning channel? To open one means you either received 500,000 sats in a one-time transaction or had more than 500,000 sats in an initial balance.

2) When you paid from Zeus to Blitz you said that Zeus showed an error, and CLN showed a successful payment but did Blitz show a successful payment?

**Reply by barfinger** [reply]
1. It looks like 40+ days ago I definitely had a >500,000 sat lightning payment that would have opened a channel. I had a channel close ~30 days ago. I received a couple more lightning payments totaling over 500,000 sats. 

2. Blitz didn't display confirmation of a successful payment.

**Reply by BlakeKaufman** [reply]
So for the initial use, you got all those funds back from the channel close event correct?

How much was the payment for? Was this a liquid-to-LN swap transaction?

**Reply by barfinger** [reply]
I sent 145,000 sats, 290,000+ went missing for ~13 days until the channel close.

[<img src="https://github.com/user-attachments/assets/d3dc5bc0-6fcc-4ce1-9922-8181047178dc" width="250px;"/>]() [<img src="https://github.com/user-attachments/assets/47c56674-ea1c-40c1-9f0a-1b006782473f" width="250px;"/>]() [<img src="https://github.com/user-attachments/assets/aec67918-5aeb-435b-8632-bd322a2a71fa" width="250px;"/>]()

**Reply by BlakeKaufman** [reply]
Well, the 145,000 SAT payment wasn't complete because there is no preimage attached. But you got all of those funds back. 


With the newest transaction that didn't complete could you walk me through what happened?

**Reply by barfinger** [reply]
I was trying to free up some liquidity and had success using Blitz. 
- I copy and pasted the Blitz invoice into Zeus 
- Tried to pay, and probably got an error about my Tor connection or it appeared to time out or stall. Zeus to StartOS CLN via Tor rarely works without a single error...
- Blitz never received anything.
- I was mobile, and didn't think anything of the common error/issue.

The successful example shows funds going direct to my node id. The missing funds go to another node id (Boltz|CLN).


<details>
<summary><b>SUCCESS EXAMPLE</b></summary>
<img src="https://github.com/user-attachments/assets/de9deabe-5e0b-4e5f-8d07-72307d2608d1" width="250px;"/> <img src="https://github.com/user-attachments/assets/87d589cc-4226-41ad-a24b-1ad9f62ac4b8" width="250px;"/> <img src="https://github.com/user-attachments/assets/610e4e1c-a56d-46fb-92f4-2fe121514bb6" width="250px;"/>
</details>


<details>
<summary><b>MISSING FUNDS EXAMPLE</b></summary>
<img src="https://github.com/user-attachments/assets/faba9935-02f5-4df1-abe1-4919c19c588e" width="250px;"/> <img src="https://github.com/user-attachments/assets/2e7c87af-e2b0-4764-bb78-fc4b5f886199" width="250px;"/> <img src="https://github.com/user-attachments/assets/db91ab24-4826-498d-84fe-22c9fabda58c" width="250px;"/> <img src="https://github.com/user-attachments/assets/25846820-e930-4d83-a26b-5be7333cad22" width="250px;"/> <img src="https://github.com/user-attachments/assets/39d710d4-31c2-4203-a59a-501c42e42f7c" width="250px;"/>
</details>

**Reply by BlakeKaufman** [reply]
If blitz didn't confirm the swap it should be refunded. Did all of those swaps fail?

One thing when sending and receiving in the same device is that blitz needs to be open to finalize the swap. Did you keep blitz open after sending the payments?

**Reply by barfinger** [reply]
> If blitz didn't confirm the swap it should be refunded. Did all of those swaps fail?

They appear to have been '_successful_'...

> One thing when sending and receiving in the same device is that blitz needs to be open to finalize the swap. Did you keep blitz open after sending the payments?

Some times I would switch back and forth. Sometimes I would run them both split screen, just in case that was the case.

Sometime in the last 40 days I had a _zero disk space_ event. It's possible I was trying to receive during that time frame. Would that be an issue?

**Reply by BlakeKaufman** [reply]
When you go into settings under "view liquid swaps" do you see anything?

Zero disk space would only make the app not work and make the payment fail. So just to be clear none of the payments you sent have either been refunded or showed up in Blitz? Also, if you try and do a 1500 sat transaction does it work?

**Reply by barfinger** [reply]
>When you go into settings under "view liquid swaps" do you see anything?

_"You have no failed liquid transactions"_

>So just to be clear none of the payments you sent have either been refunded or showed up in Blitz? 

No, and no.

>Also, if you try and do a 1500 sat transaction does it work?

I waited 6 minutes and Phoenix said pending... I forced Blitz closed, turned off my VPN, opened Blitz and my funds in Phoenix said Success. **Blitz didn't receive anything again**...

<img src="https://github.com/user-attachments/assets/9a4c6220-21bf-4e74-81cb-1b2e983c920d" width="250px;"/> <img src="https://github.com/user-attachments/assets/53b7ddcf-133a-42dc-9b12-14423f93d199" width="250px;"/>

**Reply by barfinger** [reply]
the target public key in Phoenix says it's the same `02d96eadea3d780104449aca5c93461ce67c1564e2e1d73225fa67dd3b997a6018` one that all my other missing transaction went to.

**Reply by BlakeKaufman** [reply]
It's going to the same nodeID because that's bolts node. This doesn't make any sense because I can't reproduce it. I've sent multiple swaps now and they've all confirmed on my app. What's your name in contacts? I want to try and send a payment.

**Reply by BlakeKaufman** [reply]
Can you try and put your seed into the Blockstream green app and then select the liquid network card to see if it's a Blitz Liquid wallet kit error?

**Reply by BlakeKaufman** [reply]
In debugging it looks like there is somehow an issue with the liquid address that is being generated by the liquid wallet kit in blitz. 

I am not sure what the issue is, how it randomly became a problem, why it only happens on Android, or  if it is possible to retrieve the  funds.

I'm on vacation for 5 more days and will have to look more into it then.

**Reply by barfinger** [reply]
Enjoy your vacation, I can wait to debug this later. 

Thank you so much for your help!

**Reply by BlakeKaufman** [reply]
Good news—I was able to identify and fix the bug in the latest version. When posting the swap claim transaction, the endpoint wasn’t throwing an error but simply returning one. This caused the app to mistakenly believe the transaction had succeeded, leading to the deletion of the swap information.

Unfortunately, the bad news is that the claim transaction didn’t post, which is why the funds didn’t appear. Since the swap information was deleted by the app, we no longer have it as a backup. I recommend reaching out to Boltz for assistance. While I can’t guarantee they’ll be able to help, they may be able to use the payment hash to look up the swap and post the final claim transaction.

I apologize for this issue. As the app is still in beta, there are some bugs that need to be resolved before it can be reliably used. I am willing to help in any way possible but unfortunately, there might not be much we can do.
